docs-src/prod-readiness-context/how-to-set-up-codec-server.md

```diff
@@ -1,34 +1,230 @@
 
-#### tctl
+A Codec Server is an HTTP or HTTPS Server that you create and host.
+It must be configured to use a [Payload Codec](/concepts/what-is-a-payload-codec) with the required decode logic and encryption keys.
 
-After the Codec Server is started, provide the exposed endpoint to tctl using the `--codec_endpoint` global option.
+The Codec Server is independent of the Temporal Server and decodes your encrypted payloads through endpoints.
+When you set the codec endpoint in the Temporal Web UI, the Web UI uses the remote endpoint to send encoded data to the Codec Server and receive decoded payloads from the Codec Server. See [API contract requirements](#api-contract-specifications).
+Decoded payloads are then displayed in the Workflow Execution Event History on the Web UI.
+Note that the decoded payloads are visible only to you on the Web UI; All data on the Temporal Server remains encrypted.
 
-For example, if you are running your Codec Server locally and expose port 8888 as your endpoint, you can run the following command to see the decoded output of the Workflow Execution "yourWorkflow" in the Namespace "yourNamespace".
+Since you create, operate, and manage access to your Codec Server in your controlled environment, ensure that you consider the following:
+
+- When you set your codec endpoint with your Web UI, expect your Codec Server to receive a large number of requests per Workflow Execution from the Web UI.
+- Ensure that you secure access to the decrypted data from your Codec Server. <!--Need a better way to explain this; with temporal cloud, the decrypted data is sent to the browser; there is no guarantee that the cloud ui is hosted in a particualr region etc.> need clearer way to set this expectation.-->
+- The Temporal Web UI only displays the decoded payloads received from your Codec Server in real-time; it does not store or send the data back to the Temporal Server (whether on Cloud or self-hosted Temporal Clusters).
+- You might have latencies introduced in the Web UI when sending and receiving payloads to the Codec Server.
+
+To create a Codec Server, you need the following components:
+
+- A [Payload Codec](/concepts/what-is-a-payload-codec) with the requisite keys and logic to decode your encrypted payloads.
+  You can use the Payload Codec that you applied with your Data Converter to encode your Payloads and configure it with your Codec Server.
+  However, if you are writing your Codec Server in a different SDK from the one that applies the Data Converter, ensure that your logic and keys are correctly replicated.
+- Key management infrastructure or plan for sharing your encryption keys between the Workers and your Codec Server.
+- [CORS](https://docs.google.com/document/d/16oKX7gbQWncygjv0Tjf8poW8vX-R0kxZMcrof1Y71kA/edit#bookmark=id.8boof7er3uus) configuration on the HTTP router in your Codec Server for sending and receiving requests from the Temporal Web UI.
```

Comment:
I presume we're not intending to leave a link to google docs?

Reply:
yes; I thought I'd linked these to https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS; will check and correct everywhere.